### PR TITLE
Platform/7656/clean mime

### PR DESCRIPTION
**Pull request by david-navapbc**
This PR 
   Fixes: 
   * https://github.com/CDCgov/prime-reportstream/issues/7656

Test Steps:
1. build the project and run the tests. 

## Changes
* MIME types are now in a new enum class MimeFormat
* where possible, we're using MIME constants from fhir library we're already using
* everything looking for Sender.Format or Report.Format now points to MimeFormat

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

### CODE REVIEW

**Review by thetaurean — CHANGES_REQUESTED** [review]
Changes look great! Boy was that mess getting out of hand.

Some minor comments in files and some small fixes needed:
- Props for updating docs!
- Please fix integration tests: looks like some references need updating (`TestIntegration` gradle task)
- Please finish PR template:
  - update "This PR..." text
  - add Fixes #ticketnumber 
    - Fixes #7656 
  - remove security section if not applicable
- For future branches please avoid creating per-ticket branches in `platform/`. Highly suggest naming convention: `platform/[githubname]/[ticket-number]-some-text-here`

File: prime-router/src/main/kotlin/Sender.kt [hunk 1]
```diff
@@ -108,15 +108,15 @@ abstract class Sender(
...
-    }
+//    /**
+//     * The format this sender makes submissions in
+//     */
+//    enum class Format(val mimeType: String) {
+//        CSV("text/csv"),
+//        HL7("application/hl7-v2"),
+//        FHIR("application/fhir+ndjson"),
+//        HL7_BATCH("application/hl7-v2"),
+//    }
```
[the remark this answers is not in the document]
Comment (david-navapbc) [reply prime-router/src/main/kotlin/Sender.kt:119]: oops - good catch! I will remove

File: prime-router/src/main/kotlin/Sender.kt [hunk 2]
```diff
@@ -108,15 +108,15 @@ abstract class Sender(
...
     }
 
-    /**
-     * The format this sender makes submissions in
-     */
-    enum class Format(val mimeType: String) {
-        CSV("text/csv"),
-        HL7("application/hl7-v2"),
-        FHIR("application/fhir+ndjson"),
-        HL7_BATCH("application/hl7-v2"),
```
[the remark this answers is not in the document]
Comment (david-navapbc) [reply prime-router/src/main/kotlin/Sender.kt:118]: I don't think so? The same information is being stored in the db regardless of whether or not the legacy Sender.Format was used or the new Mime.Format.

[the remark this answers is not in the document]
Comment (thetaurean) [reply prime-router/src/main/kotlin/Sender.kt:118]: The enum vals get serialized to int in the db, so I believe:

Sender.Format.HL7_BATCH would become a 4
Mime.Format.HL7_BATCH would become a 5

These could be off by 1 I can't remember if Enum->Int starts at 1 or 0.

[the remark this answers is not in the document]
Comment (david-navapbc) [reply prime-router/src/main/kotlin/Sender.kt:118]: all of the tests are passing. is there a particular issue you are concerned about that's not covered by the test rig?

**Comment by david-navapbc** [comment]
> Changes look great! Boy was that mess getting out of hand.
> 
> Some minor comments in files and some small fixes needed:
> 
>     * Props for updating docs!
> 
>     * Please fix integration tests: looks like some references need updating (`TestIntegration` gradle task)
> 
>     * Please finish PR template:
>       
>       * update "This PR..." text
>       * add Fixes #ticketnumber
>         
>         * Fixes [Clean up enumerations with duplicate mime types #7656](https://github.com/CDCgov/prime-reportstream/issues/7656)
>       * remove security section if not applicable
> 
>     * For future branches please avoid creating per-ticket branches in `platform/`. Highly suggest naming convention: `platform/[githubname]/[ticket-number]-some-text-here` e.g. `platform/thetaurean/1234-do-it`

done and done.

**Review by thetaurean — APPROVED** [review]
👍

**Review by arnejduranovic — APPROVED** [review]
Changes look good. Good job getting rid of the redundant code! Small nit: Update PR description to have the "Fixes" in the proper section below.

**Pull request merged**